lib/passport-wsfed-saml2/saml.js

```diff
@@ -119,7 +124,34 @@ SAML.prototype.validateSignature = function (xml, options, callback) {
   }
 };
 
+SAML.prototype.validateCertExpiration = function (samlAssertion) {
+  // TODO - is this version-dependent?
+  if(samlAssertion.getElementsByTagName("X509Certificate").length > 0){
+    cert = samlAssertion.getElementsByTagName("X509Certificate")[0].textContent;
```

Comment:
Is this var declared somewhere outside of scope or is it accidentally a global. You may want to assign it above the conditional and then reference the var name in the `if` so that you aren't traversing the XML twice.